app/models/course/assessment/answer/programming.rb

```diff
@@ -58,7 +58,23 @@ def assign_params(params)
 
     params[:files_attributes]&.each do |file_attributes|
       file = files.find { |f| f.id == file_attributes[:id].to_i }
-      file.content = file_attributes[:content]
+      file.content = file_attributes[:content] if file.present?
     end
   end
+
+  def create_and_update_files(params)
+    params[:files_attributes]&.each do |file_attributes|
+      file = files.find { |f| f.id == file_attributes[:id].to_i }
```

Comment:
I think you can use [`find_or_initialize_by`](http://guides.rubyonrails.org/active_record_querying.html#find-or-initialize-by) here.

Reply:
It initializes the new file to id=0, but save doesn't re-assign the id for some reason.